### PR TITLE
DRAFT: Carbonmark data charts rebased

### LINKED ISSUE #1453: [carbon-dashboard] Develop new chart type: Pie chart
Good example: Tokenized credits by bridge
See: https://recharts.org/en-US/api/PieChart

### LINKED ISSUE #1454: [carbon-dashboard] Develop new chart type: Line chart
Good example: Historical prices
See: https://recharts.org/en-US/api/LineChart

### PR DESCRIPTION
**Pull request by biwano**
## Description

- Improves Typings overall
- Improves Data transformation overall
- Improves ChartCard (Renamed from OverviewCard)
- Tokenized Credits by Bridge Card
- Historical Prices Card
- Migrates from Emotion to scss

## Related Ticket

Resolves #1453 
Resolves #1454 

## Changes

| Before  | After  |
| image  | image  |

## Checklist

- [ ] I have run `npm run build-all` without errors
- [ ] I have formatted JS and TS files with `npm run format-all`

### CODE REVIEW

File: carbonmark-data/components/charts/helpers/Configuration.tsx [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
   label: string; // label
   color: string; // color on the chart
   legendOrder?: number; // The order in which the element are displayd in the legend
 }
-export type ChartConfiguration<T> = Array<ChartConfigurationItem<T>>;
+/*
+A chart configuration is an array of configuration items that also bear data about how to query data (Q)
+*/
+export type ChartConfiguration<Q, M, C> = Array<{
+  query?: Q;
```
Comment (sprrwhwk) [inline carbonmark-data/components/charts/helpers/Configuration.tsx:11]: Why is query nullable? I can see that we throw errors if it's undefined so why not enforce it to be non-null?

File: carbonmark-data/lib/charts/helpers.ts [hunk 2]
```diff
@@ -70,8 +89,43 @@ export async function prepareDailyChartData<
...
+    record.color = chartOptions.color;
+    record.label = chartOptions.label || chartOptions.id;
+    record.quantity = dataset.quantity;
+    chartData.push(record);
+  });
+  return chartData;
+}
 
 // Common formatters
+// TODO: This goes to lib?
```
Comment (sprrwhwk) [inline carbonmark-data/lib/charts/helpers.ts:128]: I'd say it should only go to lib if it's used in other modules

File: carbonmark-data/components/charts/helpers/Configuration.tsx (hunk 1 above)
Comment (biwano) [reply carbonmark-data/components/charts/helpers/Configuration.tsx:11]: It is not mandatory for the HistocalPriceCard: https://github.com/KlimaDAO/klimadao/pull/1487/files#diff-68aa516936992eed7a72a89d9182141ae44df7e7fcaadb3ba51bb078949870e1

Truth be told, it does not felt good when I made it optional. If you think of a better design. Do not hesitate.

File: carbonmark-data/lib/charts/helpers.ts (hunk 2 above)
Comment (biwano) [reply carbonmark-data/lib/charts/helpers.ts]: Comment removed in [fd3bca0](https://github.com/KlimaDAO/klimadao/pull/1487/commits/fd3bca0fba44dae4b8fd6acd3b91bbda89540018)

**Pull request merged**